tests/src/Kernel/Migrate/OgUserMembershipD6MigrateTest.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Drupal\Tests\og\Kernel\Migrate;
+
+use Drupal\og\Entity\OgMembership;
+use Drupal\og\OgMembershipInterface;
+use Drupal\Tests\migrate_drupal\Kernel\d6\MigrateDrupal6TestBase;
+
+/**
+ * Tests organic group membership migration for users.
+ *
+ * @group og_migrate
+ */
+class OgUserMembershipD6MigrateTest extends MigrateDrupal6TestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = [
+    'text',
+    'entity_reference',
+    'filter',
+    'menu_ui',
+    'node',
+    'system',
+    'user',
+    'og',
+    'og_ui',
+    'og_migrate',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    $this->installEntitySchema('og_membership_type');
+    $this->installEntitySchema('og_membership');
+    $this->installEntitySchema('og_role');
+    $this->installEntitySchema('user');
+    $this->installEntitySchema('node');
+    $this->installConfig(['user', 'node', 'og']);
+    $this->loadFixture(__DIR__ . '/../../../fixtures/drupal6.php');
+    $this->executeMigrations([
+      'd6_filter_format',
+      'd6_user_role',
+      'd6_user',
+      'd6_node_settings',
+      'd6_node_type',
+      'd6_node',
+      'd6_og_admin_role',
```

Comment:
Moving this below d6_og_group_type reveals that default group roles need to be created in the d6_og_group_type migration. Injecting the OgRoleManager service into OgGroup destination plugin, and creating the default roles reveals some further fails.